[KB-284] Updated justification for XGboost and NGBoost in model selection (#195)
chhoumann/thesis-chemcam: report_thesis/src/sections/proposed_approach/model_selection.tex

To further bolster our selection pool, we included models from the same family as those that showed promise in the literature.

\gls{xgboost} and \gls{ngboost} both belong to the gradient boosting family, but they approach gradient boosting in distinct ways.

> \gls{xgboost} uses a more advanced gradient boosting algorithm, while \gls{ngboost} uses a more sophisticated loss function.

Reviewer: ```suggestion
\gls{xgboost} uses advanced algorithmic optimizations, while \gls{ngboost} focuses on sophisticated probabilistic loss functions.
```

Reviewer: We could say more:
```suggestion
\gls{xgboost} uses advanced algorithmic optimizations, such as regularization, tree pruning, and parallel processing, to improve performance and prevent overfitting. 
On the other hand, \gls{ngboost} focuses on sophisticated probabilistic loss functions, optimizing the natural gradient to model the entire probability distribution of the target variable, making it well-suited for tasks requiring uncertainty estimation and probabilistic forecasting.
```

I'm sure about XGBoost, but can @Pattrigue confirm the NGBoost statement?

Reviewer: My suggested changes are mostly about granularity. We want to say _something_, because there's more reason than simply trying a variation of the same algorithm.

> \gls{xgboost} and \gls{ngboost} are both in the gradient boosting family. 
> They each approach gradient boosting differently, with \gls{xgboost} using a more advanced gradient boosting algorithm and \gls{ngboost} using a more sophisticated loss function. 
> Due these factors and the limited studies regarding these in the context of \gls{libs} data, we decided to include them in our experiments.

Reviewer 2: ```suggestion
\gls{xgboost} and \gls{ngboost} both belong to the gradient boosting family, but they approach gradient boosting in distinct ways.
\gls{xgboost} uses a more advanced gradient boosting algorithm, while \gls{ngboost} uses a more sophisticated loss function.
Given these differences and the limited studies on their application to \gls{libs} data, we decided to include both in our experiments.
```
